Workaround MSVC ICE due to constexpr char* template argument

ezyang opened the pull request:
Test Plan:
Lease a Windows sandcastle https://www.internalfb.com/intern/wiki/Windows_Platform_Engineering/Leasable_VM_-_User_Guide/
and run:

```
buck build arvr/mode/win/opt //xplat/caffe2:_C_impl
```

Differential Revision: D40109191

albanD commented on torch/csrc/autograd/python_variable.cpp:
```diff
@@ -188,42 +188,18 @@ void pushPyOutToStack(
-      LOG(ERROR) << "CUDA trace hook execution failed: " << e.what();
-    }
+#define CONCRETE_TRACE_CUDA(func_name, ...) \
+  pybind11::gil_scoped_acquire gil; \
```
I know it was there before but can you move this into the `Py_IsInitialized()` scope? That might prevent some deadlock.

malfet reviewed (approved):
This PR makes me sad. We should not be using vs2017 in 2022

ezyang committed:
Workaround MSVC ICE due to constexpr char* template argument (#86288)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/86288

Test Plan:
Lease a Windows sandcastle https://www.internalfb.com/intern/wiki/Windows_Platform_Engineering/Leasable_VM_-_User_Guide/
and run:

```
buck build arvr/mode/win/opt //xplat/caffe2:_C_impl
```

Reviewed By: rkjfb, malfet

Differential Revision: D40109191

fbshipit-source-id: 42fdb36ba3b0178a6ca20527c13a9a9084ade047

ezyang commented:
@pytorchbot merge -f "github-export-checks doesn't want to pass"
